testsuite/MDAnalysisTests/analysis/test_gnm.py

```diff
@@ -0,0 +1,49 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4 fileencoding=utf-8
+#
+# MDAnalysis --- http://www.MDAnalysis.org
+# Copyright (c) 2006-2015 Naveen Michaud-Agrawal, Elizabeth J. Denning, Oliver
+# Beckstein and contributors (see AUTHORS for the full list)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+from __future__ import print_function
+
+import MDAnalysis
+import MDAnalysis.analysis.gnm
+
+from numpy.testing import (TestCase, dec, assert_equal, assert_almost_equal)
+import numpy as np
+
+import tempdir
+
+from MDAnalysisTests.datafiles import PSF, DCD
+from MDAnalysisTests import parser_not_found
+
+
+class TestGNM(TestCase):
+    @dec.skipif(parser_not_found('DCD'),
+                'DCD parser not available. Are you using python 3?')
+    def setUp(self):
+        self.tmpdir = tempdir.TempDir()
+        self.universe = MDAnalysis.Universe(PSF, DCD)
+        self.gnm = MDAnalysis.analysis.gnm.GNMAnalysis(self.universe)
```

Comment:
On 27/03/16 16:54, Richard Gowers wrote:
>
> In testsuite/MDAnalysisTests/analysis/test_gnm.py 
> <https://github.com/MDAnalysis/mdanalysis/pull/803#discussion_r57529750>:
>
> > +from numpy.testing import (TestCase, dec, assert_equal, assert_almost_equal)
> > +import numpy as np
> > +
> > +import tempdir
> > +
> > +from MDAnalysisTests.datafiles import PSF, DCD
> > +from MDAnalysisTests import parser_not_found
> > +
> > +
> > +class TestGNM(TestCase):
> > +    @dec.skipif(parser_not_found('DCD'),
> > +                'DCD parser not available. Are you using python 3?')
> > +    def setUp(self):
> > +        self.tmpdir = tempdir.TempDir()
> > +        self.universe = MDAnalysis.Universe(PSF, DCD)
> > +        self.gnm = MDAnalysis.analysis.gnm.GNMAnalysis(self.universe)
>
> Is a failure in setup an Error or a Fail?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub 
> <https://github.com/MDAnalysis/mdanalysis/pull/803/files/08ce51ff1bddb8a0906d82801e0a3983229137d9#r57529750>
>
Since it is not going to raise an AssertError, it will be an error 
wherever is the line. That makes my argument above moot. But 
semantically, I still think creating the analysis object is part of the 
test.